tests/Behat/Context/FeatureContext.php

```diff
@@ -16,39 +16,77 @@ class FeatureContext extends SilverStripeContext
     {
-        $page = $this->getSession()->getPage();
-        $versionsTable = $page->find('css', '.history-viewer .table');
-        assertNotNull($versionsTable, 'I should see a list of versions');
+        $this->getVersions();
     }
 
     /**
      * @Then I should see a list of versions in descending order
      */
     public function iShouldSeeAListOfVersionsInDescendingOrder()
     {
-        $page = $this->getSession()->getPage();
-        $versions = $page->findAll('css', '.history-viewer .table tbody tr td:nth-of-type(1)');
+        $versions = $this->getVersions();
+        $this->clickVersion(current($versions));
+    }
+
+    /**
+     * @When I click on the first version
+     */
+    public function iClickOnTheFirstVersion()
+    {
+        assertNotNull($this->getLatestVersion(), 'I should see a list of versions');
+        $this->getLatestVersion()->click();
+        // Wait for the form builder to load
+        $this->getSession()->wait(3000);
+    }
 
-        $previous = null;
+    /**
+     * @When I click on version :versionNo
+     */
+    public function iClickOnVersion($versionNo)
+    {
+        $versions = $this->getVersions(' td:first-child');
+        $desiredVersion = null;
         foreach ($versions as $version) {
             /** @var NodeElement $version */
-            if ($previous) {
-                assert($version->getValue() < $previous);
+            if ($version->getText() == $versionNo) {
+                $desiredVersion = $version;
+                break;
             }
-            $previous = $version->getValue();
         }
+        assertNotNull($desiredVersion, 'Desired version ' . $versionNo . ' was not found in the page.');
+        $this->clickVersion($desiredVersion);
     }
 
     /**
-     * @When I click on the first version
+     * Click on the given version
+     *
+     * @param NodeElement $version
      */
-    public function iClickOnTheFirstVersion()
+    protected function clickVersion(NodeElement $version)
     {
-        assertNotNull($this->getLatestVersion(), 'I should see a list of versions');
-        $this->getLatestVersion()->click();
+        $version->click();
 
         // Wait for the form builder to load
-        $this->getSession()->wait(3000);
+        $this->getSession()->wait(3000, 'window.jQuery("#Form_versionForm").length > 0');
+    }
+
+    /**
+     * Returns the versions from the history viewer list (table rows)
+     *
+     * @param string $modifier Optional CSS selector modifier
+     * @return NodeElement[]
+     */
+    protected function getVersions($modifier = '')
+    {
+        // Wait for the table to be visible
+        $this->getSession()->wait(3000, 'window.jQuery(".history-viewer .table").length > 0');
+
+        $versions = $this->getSession()
+            ->getPage()
+            ->findAll('css', '.history-viewer .table tbody tr' . $modifier);
+
+        assertNotEmpty($versions, 'I see a list of versions');
```

Comment:
Not sure an assertion belongs in a helper function like this. I'd prefer to see this just return the CSS query, and have your `iShouldSeeAListOfVersions` step do the assertions, instead of just being a public wrapper for `getVersions()`